File: tags/original/wavelets/psijni/corelib/queuearray.cpp
#include "queuearray.hpp"

queuearray::queuearray (int size) 
{ 
    N = size + 1; 
    q = new Item[size + 1];  
    head = N; 
    tail = 0;
}

int queuearray::empty()
{
    return head % N == tail;
}

void queuearray::setquadnode (quadnode &source, quadnode &result)
{
    result.xul = source[head].xul;
    result.yul = source[head].yul;
    result.xur = source[head].xur;
    result.yur = source[head].yur;
    result.xll = source[head].xll;
    result.xlr = source[head].xlr;
    result.yll = source[head].yll;
    result.ylr = source[head].ylr;
    result.energy = source[head].energy;
    result.max = source[head].max;
    result.min = source[head].min;
    result.length = source[head].length;
    result.rows = source[head].rows;
    result.cols = source[head].cols;
}

int queuearray::put (quadnode &myItem)
{
    //q[tail++] = item;
    setquadnode( myItem, q[tail]);
    tail++;
    tail = tail % N;
}

void queuearray::get (quadnode &value)
{
    head = head % N;
    setquadnode ( q[head], myItem);
    head++;
    
}

File: trunk/original/wavelets/psijni/jnixform/queuearray.java


class queuearray
{
    private quadnode q[];
    private int N, head, tail;
    public queuearray (int size) 
    { 
        N = size + 1; 
        q = new quadnode[size + 1];  
        head = N; 
        tail = 0;
    }
    
    public boolean empty ()
    {
        if ( head % N == tail ) return true;
        return false;
    }
    
    public void put (quadnode myItem)
    {
        q[tail] = myItem;
        tail++;
        tail = tail % N;
    }
    
    public quadnode pop ()
    {
        head = head % N;
        return q[head++];
    }
    
    public quadnode get () { return pop (); }
    
    
}